### PR TITLE
LF-4220: Create new add products component for soil amendment

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

Screenshot 2024-06-03 at 10 39 57 AM

- Update `plus-circle` svg stroke color to `currentColor`.
- Create a mixin `flex-column-gap` that aligns its child elements in a column direction and applies a specified gap in between. 
- Create `SmallButton` component for the "x remove" button. ([stories](http://localhost:6006/?path=/docs/components-button-smallbutton--docs))
- Update `NumberInputWithSelect` and `CompositionInput` to take `onBlur`. 
- Update `InFormButtons` to make `onConfirm` function optional. (This change will allow us to submit the request using the form's onSubmit in the future if needed. I initially considered using the form's onSubmit but ended up sending the `onConfirm` prop instead because we should not nest forms.)
- `AddSoilAmendmentProducts` ([stories](http://localhost:6006/?path=/docs/page-task-addsoilamendmentproducts--docs))
- `ProductCard` component
- `ProductDetails` component
   - Takes `productId` and renders product data. In change of saving/editing products.
   - Fields not included in the initial requirements will be added in [LF-4251](https://lite-farm.atlassian.net/browse/LF-4251).
   - The NPK validation should be updated and additional unit options should be added in LF-4251.

Jira link: https://lite-farm.atlassian.net/browse/LF-4220

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

[LF-4251]: https://lite-farm.atlassian.net/browse/LF-4251?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/webapp/src/components/Form/Button/SmallButton.tsx
Comment (Duncan-Brain) [inline packages/webapp/src/components/Form/Button/SmallButton.tsx:45]: Should we use `toLocaleLowerCase()`?

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx
Comment (Duncan-Brain) [inline packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx:52]: I think this was already here from existing add product?  I always think about: https://react.dev/reference/react/useMemo#how-to-tell-if-a-calculation-is-expensive and wonder if we can remove the memo.

**Review by Duncan-Brain — COMMENTED** [review]
Looks great too!

Only thing I was wondering was if it was expected that the react select option details would change on save for this story? (instead of 2:0:3.3 -> 4:0:3.3)
Screenshot 2024-06-04 at 4 47 57 PM

**Review by SayakaOno — COMMENTED** [review]
@Duncan-Brain thank you for reviewing!
For this story, the select option details are not supposed to be updated on save. It should be automatic once we integrate with API, so I didn't implement it.
Sorry for the confusion 🙏

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx
Comment (Duncan-Brain) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx]: I guess all of the above haha. I am just trusting the React docs that unless looping over 1000's of items its not worth it. 🤷

Comment (SayakaOno) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx]: The doc also says "There is no significant harm to doing that" so I was wondering.
I don't like that the filtering keeps happening when products don't change, but it doesn't affect the performance so I'm okay without useMemo.
I wish we could filter the product somewhere else!

Comment (kathyavini) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx]: > I wish we could filter the product somewhere else!

How about in a new selector in `productSlice`? 😄 

I think that would end up equivalent to `useMemo()` but with Redux Toolkit handling the memoization. I might be on a selector kick, but I think that would be readability without recalculating.

Comment (SayakaOno) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx]: That's what I wanted to do! I would need to update three files, so I hesitated to go for it.
But now that you mentioned, I will create a new selector 😄

**Review by Duncan-Brain — APPROVED** [review]
Looks great!

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/ProductDetails.tsx
Comment (kathyavini) [inline packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/ProductDetails.tsx:160]: Clever! So this callback will get passed to both PATCH and POST mutations? It's a bit easier to imagine with Saga but I like how it's all set up.

File: packages/webapp/src/stories/Pages/Task/AddSoilamendmentProducts/AddSoilAmendmentProducts.stories.tsx
Comment (kathyavini) [inline packages/webapp/src/stories/Pages/Task/AddSoilamendmentProducts/AddSoilAmendmentProducts.stories.tsx:1]: There's a capitalization error in the folder name if you want to correct that.

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx
Comment (kathyavini) [inline packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx:122]: Is it correct that a farm not pursuing certification can add an infinite number of empty product cards since the `invalidProducts` array is empty?

Screenshot 2024-06-14 at 10 05 35 AM

**Review by kathyavini — APPROVED** [review]
🙇 🙇 🙇  Very sorry for the very late review. For yesterday's delay at least I will also blame Storybook as I had crashes on this branch all afternoon and I am still not sure what resolved them. That did however force to me carefully read the code instead of testing which I enjoyed -- these are very well-written components!!

Please merge 🙏  Since you are continuing with this component any comments that you might want to address can go on the next ticket.

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx
Comment (SayakaOno) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx:29]: 😨!!! Thank you for catching, it's surprising for me too!
I'll import `ProductId` type!

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx
Comment (SayakaOno) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx:122]: No, it's a bug! The button shouldn't be enabled if a product hasn't been selected!
Thank you :)

File: packages/webapp/src/stories/Pages/Task/AddSoilamendmentProducts/AddSoilAmendmentProducts.stories.tsx
Comment (SayakaOno) [reply packages/webapp/src/stories/Pages/Task/AddSoilamendmentProducts/AddSoilAmendmentProducts.stories.tsx:1]: Oh, thank you for your careful eye!

**Review by SayakaOno — COMMENTED** [review]
Thank you @kathyavini for reviewing!

There are some errors I don't like, so I'll update the PR.
I'm sorry to dismiss your approvals!

File: packages/webapp/src/components/Form/Button/SmallButton.tsx
Comment (kathyavini) [reply packages/webapp/src/components/Form/Button/SmallButton.tsx:42]: Oh! Thank you for keeping the release merge conflicts in mind. I would have cried a bit when I realized 🤣

**Comment by SayakaOno** [comment]
Major changes I made after the approvals are to:
- fix the "Add another product" button so it's not enabled if a product hasn't been selected (for farms not pursuing certification).
- adjust the position of the remove button on mobile view (I may need to adjust breakpoints later).
- display remove buttons when the number of products is greater than one, based on the discussion in grooming.

Thank you for re-reviewing!

**Review by kathyavini — APPROVED** [review]
Looks great, thank you!

**Pull request merged**
